app/src/main/java/co/devhack/compose/southamericanqualifiers/TeamsActivity.kt

```diff
@@ -0,0 +1,60 @@
+package co.devhack.compose.southamericanqualifiers
+
+import android.os.Bundle
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.Text
+import androidx.compose.foundation.lazy.LazyColumnFor
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.State
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.platform.setContent
+import androidx.ui.tooling.preview.Preview
+import co.devhack.compose.southamericanqualifiers.model.Team
+import co.devhack.compose.southamericanqualifiers.ui.SouthAmericanQualifiersTheme
+import co.devhack.compose.southamericanqualifiers.viewmodels.TeamsViewModel
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        setContent {
+            SouthAmericanQualifiersTheme {
+                val viewModel = viewModels<TeamsViewModel>().value
+                val teamsState = viewModel.teams.observeAsState()
+                TeamsScreen(teamsState)
+            }
+        }
+    }
+
+    @Composable
+    fun TeamsScreen(teamsState: State<TeamsViewModel.State?>) {
+        Surface(color = MaterialTheme.colors.background) {
+            when (teamsState.value) {
+                is TeamsViewModel.State.Loading -> Text("Loading...")
+                is TeamsViewModel.State.Ready -> TeamsList((teamsState.value as TeamsViewModel.State.Ready).teams)
```

Comment:
minor: we can avoid manual typecasting here by declaring a variable scoped to the `when` expression

```kotlin
when (val state = teamsState.value) {
    is TeamsViewModel.State.Loading -> Text("Loading...")
    is TeamsViewModel.State.Ready -> TeamsList(state.teams)
}
```

Reply:
Agree!